presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -173,6 +174,78 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateErrorTask(
   return std::make_unique<TaskInfo>(info);
 }
 
+// If the plan fragment contains operator that can spill and spilling is
+// enabled, created the spilling directory for the task.
+static std::string prepareTaskSpillDirectory(
+    const velox::core::QueryCtx* queryCtx,
+    const TaskId& taskId,
+    const velox::core::PlanFragment& planFragment) {
+  if (not queryCtx->queryConfig().spillEnabled()) {
+    return "";
+  }
+
+  const auto baseSpillPath = SystemConfig::instance()->spillerSpillPath();
+  if (baseSpillPath.empty()) {
+    return "";
+  }
+
+  // Functors to check if a given node can spill.
+  const auto hasSpillingHashAggrFunc = [](const core::PlanNode* node) {
+    if (node->name() == "Aggregation") {
+      if (const auto* aggregationNode =
+              dynamic_cast<const core::AggregationNode*>(node)) {
+        if (aggregationNode->isFinal() || aggregationNode->isSingle()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  };
+  const auto hasSpillingOrderByFunc = [](const core::PlanNode* node) {
+    return (node->name() == "OrderBy");
+  };
+  const auto hasSpillingHashJoinFunc = [](const core::PlanNode* node) {
+    return (node->name() == "HashJoin");
+  };
+
+  // Determine if this task can spill at all, to avoid creating a folder for
+  // nothing. Looks for particular nodes for that.
+  const auto* topNode = planFragment.planNode.get();
+  if ((queryCtx->queryConfig().aggregationSpillEnabled() &&
+       core::PlanNode::findFirstNode(topNode, hasSpillingHashAggrFunc)) ||
```

Comment:
Can you define a method in core::PlanNode::isSpillable which is also easier to maintain in the future?
Then the traversal callback would be simple as
```
const auto findSpillableNode = [](const core::PlanNode* node) {
    return node->isSpillable();
  };
```

Reply:
I thought about it, but it won't be useful, because we need to look for a particular node in tandem with a particular flag.
Like here we check "aggregationSpillEnabled()" and need to look only for aggregation nodes, not for any other.
In that scenario usefulness of the isSpillable() is zero.